addons/html_editor/static/src/others/embedded_components/core/table_of_content/table_of_content.xml

```diff
@@ -1,7 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <templates id="template" xml:space="preserve">
 <t t-name="html_editor.EmbeddedTableOfContent">
-    <div class="o_embedded_toc_content">
+    <div class="d-flex align-items-center cursor-pointer user-select-none" t-on-click="toggleToc">
+        <div class="o_embedded_toc_label px-2 py-1 text-break">
+            Table of Contents
+            <i t-attf-class="align-self-center fa fa-fw #{this.state.open ? 'fa-caret-down' : 'fa-caret-right'}"/>
+        </div>
+    </div>
+    <div class="o_embedded_toc_content" t-attf-style="#{!this.state.open or !state.toc.headings.length ? 'display: none' : 'margin-bottom: 5px'}">
```

Comment:
CSS rule adds the margin on left and right. The purpose of using this way is to add a 5px space after the content. This is done by adding margin bottom when opened and making the parent having overflow set as auto or hidden. (else margin of child will have no effect on parent)

This can also be done using the padding but that will make the "Table of Contents" appear slightly above the center.